sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+        >>> b.bc_deflection = [(0, 2)]
+        >>> b.bc_slope = [(0, 1)]
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+
+        Here the deflection of the beam should be ``2`` at ``0``.
+        Similarly, the slope of the beam should be ``1`` at ``0`` and
+        there are two boundary conditions for bending moment. The bending
+        moment of the beam should be ``4`` at ``0`` and ``0`` at ``4``
+        """
+        return self._boundary_conditions
+
+    @property
+    def bc_moment(self):
+        return self._boundary_conditions['moment']
+
+    @bc_moment.setter
+    def bc_moment(self, m_bcs):
+        self._boundary_conditions['moment'] = m_bcs
+
+    @property
+    def bc_slope(self):
+        return self._boundary_conditions['slope']
+
+    @bc_slope.setter
+    def bc_slope(self, s_bcs):
+        self._boundary_conditions['slope'] = s_bcs
+
+    @property
+    def bc_deflection(self):
+        return self._boundary_conditions['deflection']
+
+    @bc_deflection.setter
+    def bc_deflection(self, d_bcs):
+        self._boundary_conditions['deflection'] = d_bcs
+
+    def apply_load(self, value, start, order, end=None):
+        """
+        This method adds up the loads given to a particular beam object.
+
+        Parameters
+        ==========
+        value :
+            The magnitude of an applied load.
+        start :
+            The starting point of the applied load.
+        end :
+            An optional argument that can be used if the load have an end point
+            within the length of the beam.
+        order :
+            The order of the applied load.
+
+            For Moments, order = -2
+            For Pointloads, order = -1
+            For constant distributed load, order = 0
+            For Ramp load, order = 1
+            For Parabolic Ramp, order = 2
+            ... so on.
+
+        Examples
+        ========
+        There is a beam of length 4 meters. A moment of magnitude 3 Nm is
+        applied in the clockwise direction at the starting point of the beam.
+        A pointload of magnitude 4 N is applied from the top of the beam at
+        2 meters from the starting point and a parabolic ramp load of magnitude
+        2 Nm/m is applied below the beam starting from 2 meters to 3 meters
+        away from the starting point of the beam.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_load(value=-3, start=0, order=-2)
+        >>> b.apply_load(value=4, start=2, order=-1)
+        >>> b.apply_load(value=-2, start=2, order=2, end = 3)
+        >>> b.load
+        -3*SingularityFunction(x, 0, -2) + 4*SingularityFunction(x, 2, -1) - 2*SingularityFunction(x, 2, 2)
+            + 2*SingularityFunction(x, 3, 0) + 2*SingularityFunction(x, 3, 2)
```

Comment:
It doesn't seem like you have the reaction loads or any kind of supports defined.